model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -402,26 +427,104 @@ def _parse_outputs(self, metrics):
         ]:
             os.remove(perf_config['latency-report-file'])
 
-    def _extract_metrics_from_row(self, requested_metrics, row_metrics):
+    def _extract_metrics_from_row(self, requested_metrics: List[Record],
+                                  row_metrics: Dict[str, str]) -> List[Record]:
         """ 
         Extracts the requested metrics from the CSV's row and creates a list of Records
         """
-        perf_records = []
+        perf_records: List[Record] = []
+
+        perf_metrics, perf_values = self._find_requested_perf_metrics(
+            requested_metrics, row_metrics)
+        perf_records = self._add_perf_metrics_to_records(
+            perf_metrics, perf_values, perf_records)
+
+        gpu_metrics, gpu_values = self._find_requested_gpu_metrics(
+            requested_metrics, row_metrics)
+        perf_records = self._add_gpu_metrics_to_records(gpu_metrics, gpu_values,
+                                                        perf_records)
+
+        return perf_records
```

Comment:
The consuming code is going to have to split these back out to be perf_records and gpu_records. It might be best to just store them separately

Reply:
That very well might be our end solution, but that would break upstream code, which I'd rather not touch as part of this story. It seems a trivial enough change to make in the follow on story if it's needed.